enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,398 @@
+1. When a user runs `oc adm upgrade status`, the `oc` client reads the `UpdateStatus` and reports that no update is happening.
+1. The user triggers the update.
+1. The USC monitors `ClusterVersion`, `ClusterOperator`, `MachineConfigPool`, and `Node` resources and reflects the state of the update through the `UpdateStatus` resource via a set of status and health insights.
+1. When the user runs `oc adm upgrade status`, the client reads the `UpdateStatus` and uses status insights to convey progress and health insights to convey issues the admin needs to address.
+
+### API Extensions
+
+Introduce a new Cluster-scoped `UpdateStatus` CRD to convey update status information. For now, its `.spec` is empty but can be used in the future to configure the desired form or content of the information surfaced in status.
+
+The `status` subresource content is the primary value of the enhancement. The status and health information about the update are expressed through the `status` subresource.
+
+Introduce a new `update.openshift.io` group for `UpdateStatus`. There is no suitable existing OpenShift API group for where `UpdateStatus` would fit. The `ClusterVersion` CRD, which is used to convey some information about the update process, is in the `config.openshift.io` group. `UpdateStatus` provides no configuration capabilities, so `config.openshift.io` is a poor fit. The new group is well-suited to contain some of the other update-related APIs needed for further incoming features like maintenance windows or update pre-checks.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The enhancement allows adoption in HCP as an extension. The API and controller will be extended to handle the resources that represent the control plane and worker pools (`HostedCluster` and `NodePool`, respectively) and surface update progress through status insights for these resources.
+
+The Update Status Controller will run as a part of the hosted control plane. Unlike in Standalone, where it is a CVO operand, USC will be managed by the Control Plane Controller in HCP, where the CVO does not have access to management cluster API and cannot manage resources there. USC needs to access the management cluster API layer to manage the status-reporting API (which targets cluster administrators tasked with updating clusters) and needs to report insights about both the hosted cluster and management cluster resources. It will also need access to the hosted cluster API layer to report on resources present there.
+
+Because `UpdateStatus` API is cluster-scoped, it is not suitable for HCP use, and we will need to introduce a namespaced HCP-specific flavor, such as `HostedClusterUpdateStatus`. HCP-specific API variant will also need to be able to refer to resources in two API surfaces (management and hosted clusters), while the `UpdateStatus` API can rely on all resources being in the same cluster and use simple resource references.
```

Comment:
This section talks about an `HostedClusterUpdateStatus`. In this proposal, is this resource expected to contain also the NodePool status, or there will be N `NodePoolUpdateStatus`, one for each NodePool?
I think it would be better to be clear on this point.
Option 2 is more aligned with HyperShift topology IMO

Reply:
The `UpdateStatus` is more like a container for insights, there's exactly one for a cluster, so in HCP it needs to be namespaced and needs to have a different name - `HostedCluster...` here refers more to a "hosted cluster" as an entity, not `HostedCluster` as a resource.

The actual status tied to a resource that represents the cluster would be exposed through a status insight, so you'd have a `HostedClusterStatusInsight` (=HCP analogue of the `ClusterVersionStatusInsight` on Standalone), and also a `NodePoolStatusInsight` per each NodePool (=HCP analogues of the `MachineConfigPoolStatusInsight` on Standalone).